PWGHF/D2H/TableProducer/dataCreatorDV0Reduced.cxx

```diff
@@ -0,0 +1,393 @@
+      if (grpo == nullptr) {
+        LOGF(fatal, "Run 3 GRP object (type o2::parameters::GRPMagField) is not available in CCDB for run=%d at timestamp=%llu", bc.runNumber(), bc.timestamp());
+      }
+      o2::base::Propagator::initFieldFromGRP(grpo);
+      // setMatLUT only after magfield has been initalized
+      o2::base::Propagator::Instance()->setMatLUT(lut);
+      runNumber = bc.runNumber();
+    }
+
+    for (const auto& candD : candsD) {
+      // initialize variables depending on decay channel
+      bool fillHfCandD = false;
+      float invMassD;
+      float massD;
+      float massV0;
+      std::array<float, 3> pVecD;
+      std::array<float, 3> secondaryVertexD;
+      std::array<int, 3> prongIdsD;
+      int8_t v0_type;
+      int8_t d_type;
+
+      if constexpr (std::is_same<C, CandDstarFiltered>::value) {
+        if (candD.signSoftPi() > 0)
+          invMassD = candD.invMassDstar() - candD.invMassD0();
+        else
+          invMassD = candD.invMassAntiDstar() - candD.invMassD0Bar();
+        massD = MassDStar;
+        pVecD = candD.pVector();
+        secondaryVertexD[0] = candD.xSecondaryVertexD0();
+        secondaryVertexD[1] = candD.ySecondaryVertexD0();
+        secondaryVertexD[2] = candD.zSecondaryVertexD0();
+        prongIdsD[0] = candD.prong0Id();
+        prongIdsD[1] = candD.prong1Id();
+        prongIdsD[2] = candD.prongPiId();
+        v0_type = V0_type::K0s;
+        d_type = candD.signSoftPi() * D_type::Dstar;
+        massV0 = MassK0Short;
+      } else if constexpr (std::is_same<C, CandsDplusFiltered>::value) {
+        switch (DecayChannel) {
+          case DecayChannel::DsStar2ToDplusK0s:
+            invMassD = hfHelper.invMassDplusToPiKPi(candD);
+            massD = MassDPlus;
+            pVecD = candD.pVector();
+            secondaryVertexD[0] = candD.xSecondaryVertex();
+            secondaryVertexD[1] = candD.ySecondaryVertex();
+            secondaryVertexD[2] = candD.zSecondaryVertex();
+            prongIdsD[0] = candD.prong0Id();
+            prongIdsD[1] = candD.prong1Id();
+            prongIdsD[2] = candD.prong2Id();
+            v0_type = V0_type::K0s;
+            d_type = candD.sign() * D_type::Dplus;
+            massV0 = MassK0Short;
+            break;
+
+          case DecayChannel::Xc3055ToDplusLambda:
+            invMassD = hfHelper.invMassDplusToPiKPi(candD);
+            massD = MassDPlus;
+            pVecD = candD.pVector();
+            secondaryVertexD[0] = candD.xSecondaryVertex();
+            secondaryVertexD[1] = candD.ySecondaryVertex();
+            secondaryVertexD[2] = candD.zSecondaryVertex();
+            prongIdsD[0] = candD.prong0Id();
+            prongIdsD[1] = candD.prong1Id();
+            prongIdsD[2] = candD.prong2Id();
+            v0_type = candD.sign() * V0_type::Lambda;
+            d_type = candD.sign() * D_type::Dplus;
+            massV0 = MassLambda0;
+            break;
```

Comment:
These two should go together based on the comment above; you can check if the V0 is compatible with a Lambda then you compute the D+Lambda mass, if it is compatible with the K0S you compute the D+K0S mass (depending on the cuts, it might be both)

For this we also need to distinguish K0 and Lambda (see below)